integration_test/src/main.rs

```diff
@@ -277,6 +278,55 @@ fn test_get_balances_generate_to_address(cl: &Client) {
     }
 }
 
+fn test_generate_block(cl: &Client) {
+    let sk = PrivateKey {
+        network: Network::Regtest,
+        inner: secp256k1::SecretKey::new(&mut secp256k1::rand::thread_rng()),
+        compressed: true,
+    };
+    let addr = Address::p2wpkh(&sk.public_key(&SECP), Network::Regtest).unwrap();
+
+    let options = json::ListUnspentQueryOptions {
+        minimum_amount: Some(btc(2)),
+        ..Default::default()
+    };
+    let unspent = cl.list_unspent(Some(6), None, None, None, Some(options)).unwrap();
+    let unspent = unspent.into_iter().nth(0).unwrap();
+
+    let tx = Transaction {
+        version: 1,
+        lock_time: PackedLockTime::ZERO,
+        input: vec![TxIn {
+            previous_output: OutPoint {
+                txid: unspent.txid,
+                vout: unspent.vout,
+            },
+            sequence: Sequence::MAX,
+            script_sig: Script::new(),
+            witness: Witness::new(),
+        }],
+        output: vec![TxOut {
+            value: (unspent.amount - *FEE).to_sat(),
+            script_pubkey: addr.script_pubkey(),
+        }],
+    };
+
+    let input = json::SignRawTransactionInput {
+        txid: unspent.txid,
+        vout: unspent.vout,
+        script_pub_key: unspent.script_pub_key,
+        redeem_script: None,
+        amount: Some(unspent.amount),
+    };
+    let res = cl.sign_raw_transaction_with_wallet(&tx, Some(&[input]), None).unwrap();
+    assert!(res.complete);
+
+    let raw_tx = bitcoincore_rpc::MineableTx::RawTx(res.transaction().unwrap());
+    let result = cl.generate_block(&cl.get_new_address(None, None).unwrap(), &[raw_tx]).unwrap();
```

Comment:
Should we also put a tx in the mempool and use its txid here to test both cases of the enum?

Otherwise ACK.

Reply:
done